pkg/broker/handler/processors/deliver/processor.go

```diff
@@ -101,41 +104,69 @@ func (p *Processor) Process(ctx context.Context, event *event.Event) error {
 	}
 
 	// Forward the event copy that has hops removed.
-	resp, res := p.DeliverClient.Request(cecontext.WithTarget(dctx, target.Address), copy)
-	if !protocol.IsACK(res) {
+	if err := p.deliver(dctx, target, broker, (*binding.EventMessage)(&copy), hops); err != nil {
 		if !p.RetryOnFailure {
-			return fmt.Errorf("target delivery failed: %v", res.Error())
+			return err
 		}
 
-		logging.FromContext(ctx).Warn("target delivery failed", zap.String("target", tk), zap.String("error", res.Error()))
+		logging.FromContext(ctx).Warn("target delivery failed", zap.String("target", tk), zap.Error(err))
```

Comment:
Desugar or switch to `knative.dev/eventing/pkg/logging`.

This mistake is repeated a few times in the file, so I suggest switching the import.

Reply:
I'm not sure that the logging throughput is high enough here for desugaring to be useful. Profiling data shows very little time is spent in the logger and desugaring may actually make things slower because we recreate a desugared logger on every call instead of storing the logger.

Reply:
Resolved offline that desugared logger is still worth using for safety reasons.